frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/EditCamperCard.tsx

```diff
@@ -0,0 +1,409 @@
+          camperIndex,
+          data: initialCamper.firstName,
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CAMPER,
+          field: "lastName",
+          camperIndex,
+          data: initialCamper.lastName,
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CAMPER,
+          field: "age",
+          camperIndex,
+          data: initialCamper.age,
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CAMPER,
+          field: "allergies",
+          camperIndex,
+          data: initialCamper.allergies,
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CAMPER,
+          field: "specialNeeds",
+          camperIndex,
+          data: initialCamper.specialNeeds,
+        });
+
+        setEditing(false);
+
+        toast({
+          description: `Edits have been discarded.`,
+          status: "success",
+          duration: 3000,
+          isClosable: false,
+          variant: "subtle",
+        });
+        onClose();
+      },
+    });
+
+    return (
+      <>
+        <Button
+          color="text.critical.100"
+          variant="outline"
+          onClick={onOpen}
+          colorScheme="red"
+          w={{ sm: "80px", lg: "100px" }}
+          h={{ sm: "30px", lg: "40px" }}
+        >
+          <Text textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}>
+            Cancel
+          </Text>
+        </Button>
+        {deleteCamperModal}
+      </>
+    );
+  };
+
+  return (
+    <Box boxShadow="lg" rounded="xl" borderWidth={1} width="100%" mt={2} mb={2}>
+      <Box backgroundColor="#FFFFFF" rounded="xl">
+        <Heading textStyle="displayLarge">
+          <Flex py={6} px={{ sm: "5", lg: "20" }} alignItems="center">
+            <Text textStyle={{ sm: "xSmallBold", lg: "displayLarge" }}>
+              {camper.firstName} {camper.lastName}
+            </Text>
+            <Spacer />
+            <Button
+              variant="secondary"
+              textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}
+              disabled={editing}
+              w={{ sm: "80px", lg: "100px" }}
+              h={{ sm: "30px", lg: "40px" }}
+              onClick={() => setEditing(true)}
+            >
+              Edit
+            </Button>
+          </Flex>
+        </Heading>
+        <Divider borderColor="border.secondary.100" />
+      </Box>
+
+      <Box
+        zIndex={0}
+        backgroundColor="#FFFFFFAA"
+        borderRadius="0px 0px 10px 10px"
+        _hover={{ cursor: editing ? "auto" : "not-allowed" }}
+      >
+        <Box zIndex={editing ? 1 : -1} position="relative">
+          <Box px={{ sm: "5", lg: "20" }}>
+            <Wrap pt={7}>
+              <WrapItem width={{ sm: "100%", md: "45%", lg: "30%" }}>
+                <FormControl isInvalid={isFirstNameInvalid}>
+                  <FormLabel>
```

Comment:
Could we make the FormLabel it's own component in this file called `EditCamperCardFormLabel` which would just return this but you can pass in the label title. I want to reduce code duplication in this file so it's more readable

Reply:
Done.